runners/reference/job-server/src/test/java/org/apache/beam/runners/reference/job/ReferenceRunnerJobServiceTest.java

```diff
@@ -85,8 +96,39 @@ public void testPrepareJob() throws Exception {
     File foo = writeTempFile("foo", "foo, bar, baz".getBytes());
     File bar = writeTempFile("spam", "spam, ham, eggs".getBytes());
     stager.stage(ImmutableList.<File>of(foo, bar));
+    List<byte[]> tempDirFiles = readFlattendFiles(temp.getRoot());
```

Comment:
This doesn't seem right, your comparing the files you created via writeTempFile against itself, shouldn't you be comparing against what was staged?

Also, it would be best if you could ensure that the artifact coincides with the expected one that was uploaded by name.

Reply:
The former is done.

I'm not wholly in agreement on the latter - this is already kind of a skeevy check, as to verify proper functioning we need to actually build the client for the retrieval service and verify that we can re-acquire the original files, but there's no reason the artifact staging service needs to represent the staged files in a way that is meaningful to the clients on either end, so long as it can reconstruct the original files.

Reply:
When the artifact download service becomes ready, it seems valuable to make that integration test.

Reply:
Strongly agreed